src/main/kotlin/nomic/api/GamesEndpoint.kt

```diff
@@ -34,6 +38,54 @@ class GamesEndpoint(val gameDomain: GameDomain) {
         return ResponseEntity(ResponseFormat(true, HttpStatus.CREATED, "Game Created"), HttpStatus.CREATED)
     }
 
+    /**
+     * This endpoint listens on `api/game/list`, allowing callers to paginate through the list
+     * of games that the authenticated user has created. It accepts a size and an optional offset
+     * parameter through [ListGamesApiRequestModel].
+     *
+     * @param[input] The model encapsulating the endpoint parameters
+     * @param[user] The user whose games will be listed, injected by the authentication framework.
+     * @return A Spring entity representing the response that gets serialized into JSON
+     */
+    @GetMapping("list")
+    fun listGames(
+        @RequestBody input: ListGamesApiRequestModel,
+        @AuthenticationPrincipal authUser: EndUser
+    ): ResponseEntity<*> {
+        if (0U >= input.size || 100U < input.size) {
+            return ResponseEntity(
+                ResponseFormat(
+                    false,
+                    HttpStatus.BAD_REQUEST,
+                    "You must request at least 1 game and at most 100 games."
+                ),
+                HttpStatus.BAD_REQUEST
+            )
+        }
+
+        val games = gameDomain.listGames(authUser, input.size, input.offset)
+
+        if (games.isEmpty) {
+            return ResponseEntity(
+                ResponseFormat(
+                    false,
+                    HttpStatus.NOT_FOUND,
+                    "Invalid offset"
+                ),
+                HttpStatus.NOT_FOUND
+            )
```

Comment:
Similar idea is there not a way to let the globalExceptionHandler solve this problem instead of having the code exist twice?